### ISSUE TITLE
Calibration upload 48588f

### ISSUE THREAD

**Issue by bronger** [lensfun/lensfun#1933]
Calibration images were uploaded to the directory that starts with “`48588f_`”.

Please read the [workflow description](https://github.com/lensfun/lensfun/blob/master/tools/calibration_webserver/workflow.rst) for further instructions about the calibration.

Additional comments by the uploader:

> Coming here to add some data for the issue #1926. I took some images of buildings on infinity, mainly with the super wide angle lens. Keep in mind it is not very sharp by default so I can't get the images any sharper. Also, there are a few images from the normal wide angle camera but the distortion on it is negligible so if you do with to correct the little that is there (and not just stick with correcting the pesky vignetting), I encourage you to fall back on the test chart shots I have in the first upload. I think that one provides a better idea as the test chart is clinically straight, unlike buildings.
> 
> I hope these images are good enough. It's tough with an extremely wide lens like this.

**Comment by tuxfanx** [comment]
Here you can find the profile for the S21 Ultrawide lens. I had to use the test images from the first upload. Unfortunately, the lines on it don't go all the way to the edges of the picture either. TCA could not be corrected because of the blur and the vignetting correction is at least questionable.
Perhaps you would like to test the calibration data again and report back whether you think they are good enough.
In my opinion, creating a profile for the wide-angle lens makes no sense. I have the impression that the software has already corrected this.
[Samsung-S21-UW.zip](https://github.com/lensfun/lensfun/files/9956174/Samsung-S21-UW.zip)

**Comment by Faandyy** [comment]
> Here you can find the profile for the S21 Ultrawide lens. I had to use the test images from the first upload. Unfortunately, the lines on it don't go all the way to the edges of the picture either. TCA could not be corrected because of the blur and the vignetting correction is at least questionable. Perhaps you would like to test the calibration data again and report back whether you think they are good enough. In my opinion, creating a profile for the wide-angle lens makes no sense. I have the impression that the software has already corrected this. [Samsung-S21-UW.zip](https://github.com/lensfun/lensfun/files/9956174/Samsung-S21-UW.zip)

Thank you very much, I'll try it out as soon as I get home. Can I ask what made the images from the second upload unfitting, I tried quite hard to find buildings with straight lines. Would it help if I took a few more test shots of the grid?

I don't believe Samsung software corrects all the distortion natively even when you shoot in jepg, the reason I'm doing this whole thing. Plus there's no way to apply any potential correction profiles from Samsung on RAW files.

Can I also know what makes the vignetting correction "questionable"? Did I make a mistake somewhere?

**Comment by tuxfanx** [comment]
> > Here you can find the profile for the S21 Ultrawide lens. I had to use the test images from the first upload. Unfortunately, the lines on it don't go all the way to the edges of the picture either. TCA could not be corrected because of the blur and the vignetting correction is at least questionable. Perhaps you would like to test the calibration data again and report back whether you think they are good enough. In my opinion, creating a profile for the wide-angle lens makes no sense. I have the impression that the software has already corrected this. [Samsung-S21-UW.zip](https://github.com/lensfun/lensfun/files/9956174/Samsung-S21-UW.zip)
> 
> Thank you very much, I'll try it out as soon as I get home. Can I ask what made the images from the second upload unfitting, I tried quite hard to find buildings with straight lines. Would it help if I took a few more test shots of the grid?
>
**Which of your new pictures do you think corresponds to the instructions given on the websites?
Most of the pictures are taken in portrait format which is not so good. In the remaining images, the lines are either not continuous from image edge to image edge or the lines are in unfavorable positions. Gutters and brick walls are not suitable.** 
> 
> I don't believe Samsung software corrects all the distortion natively even when you shoot in jepg, the reason I'm doing this whole thing. Plus there's no way to apply any potential correction profiles from Samsung on RAW files.
> 
> Can I also know what makes the vignetting correction "questionable"? Did I make a mistake somewhere?
>
**Look at your image for vignetting calibration in your raw development software with the correction profile switched on. A bright spot remains in the center of the image. A homogeneously illuminated area should normally appear after the correction. May be the photographed area wasn't really evenly lit when you took the picture.**

**Comment by Faandyy** [comment]
> > > Here you can find the profile for the S21 Ultrawide lens. I had to use the test images from the first upload. Unfortunately, the lines on it don't go all the way to the edges of the picture either. TCA could not be corrected because of the blur and the vignetting correction is at least questionable. Perhaps you would like to test the calibration data again and report back whether you think they are good enough. In my opinion, creating a profile for the wide-angle lens makes no sense. I have the impression that the software has already corrected this. [Samsung-S21-UW.zip](https://github.com/lensfun/lensfun/files/9956174/Samsung-S21-UW.zip)
> > 
> > 
> > Thank you very much, I'll try it out as soon as I get home. Can I ask what made the images from the second upload unfitting, I tried quite hard to find buildings with straight lines. Would it help if I took a few more test shots of the grid?
> 
> **Which of your new pictures do you think corresponds to the instructions given on the websites? Most of the pictures are taken in portrait format which is not so good. In the remaining images, the lines are either not continuous from image edge to image edge or the lines are in unfavorable positions. Gutters and brick walls are not suitable.**
> 
> > I don't believe Samsung software corrects all the distortion natively even when you shoot in jepg, the reason I'm doing this whole thing. Plus there's no way to apply any potential correction profiles from Samsung on RAW files.
> > Can I also know what makes the vignetting correction "questionable"? Did I make a mistake somewhere?
> 
> **Look at your image for vignetting calibration in your raw development software with the correction profile switched on. A bright spot remains in the center of the image. A homogeneously illuminated area should normally appear after the correction. May be the photographed area wasn't really evenly lit when you took the picture.**

I apologise for not providing good data then. You should not have bothered with the correction profiles. But truth is, there are not many examples provided on the sites so I honestly don't have a good idea of what exactly the pic needs to be.

I don't know of any buildings that would have long continuous lines and honestly that's an extremely specific object to require of someone to photograph.

The bright spot in the middle could be a hotspot. The wide angle lens is sort of trash and relies a lot on a ton of processing done by the phone by default. I'm surprised Samsung even allowed us to get the raw data. I obtained the vignetting images by putting a sheet of PTFE in front of the lens. I use it as a white calibration subject for my full spectrum photography. It's a very good diffuser and diffuses all light equally.

**Comment by tuxfanx** [comment]
You can test the vignetting correction. Just take a few photos of evenly lit areas and then check if the vignetting is compensated correctly. 
I would like to add your 17mm TTArtisan lens to the database. It would be nice if you could provide some more pictures.

![Samsung_Galaxy_S21_ultrawide_camera](https://user-images.githubusercontent.com/95431163/200555896-2426bfe0-b8d9-4e85-b4ca-afad89c0987c.jpg)
**So it would be a good image for distortion correction**

For a TCA correction high-contrast images with an aperture of 5.6 or 8 would be good  (e.g. a window front with white frames and dark panes from the center of the image to the corners of the image).

**Comment by Faandyy** [comment]
Thank you for your interest! I will definitely add this on my list of things to do as it would benefit not only myself but also many people around the world that use the TTArtisan 17mm f/1.4.

I tried out your correction profiles for the ultrawide lens and they are pretty much perfect, thank you!

https://imgur.com/MjxtiLs

I tried applying your correction profile to the vignette test image and proceded to normalize the histogram somewhat, I believe the spots are simply an effect of either imperfections in the lens or the fact that the diffuser is partially transparent and the lens picked up the clouds. Alternatively it could also be mildly out of focus imperfections in the PTFE sheet I used as a diffuser. The lens is extremely wide and sort of dark for the sensor size, so the depth of field is very large. I wouldn't worry about it. I tried it on some of the RAW files I've taken in the past and it works great. I can get great use of this and I'm sure anyone else with the same phone can.
